src/main/java/com/openshift/internal/restclient/model/deploy/DeploymentRequest.java

```diff
@@ -0,0 +1,62 @@
+/*******************************************************************************
+ * Copyright (c) 2016 Red Hat, Inc.
+ * Distributed under license by Red Hat, Inc. All rights reserved.
+ * This program is made available under the terms of the
+ * Eclipse Public License v1.0 which accompanies this distribution,
+ * and is available at http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     Red Hat, Inc. - initial API and implementation
+ ******************************************************************************/
+package com.openshift.internal.restclient.model.deploy;
+
+import java.util.Map;
+
+import org.jboss.dmr.ModelNode;
+
+import com.openshift.internal.restclient.model.KubernetesResource;
+import com.openshift.restclient.IClient;
+import com.openshift.restclient.model.deploy.IDeploymentRequest;
+
+public class DeploymentRequest extends KubernetesResource implements IDeploymentRequest {
```

Comment:
We didnt do this with BuildRequest but probably should have.  I tried to separate by:  'KubernetesResource' is for top level objects that actually have an endpoint vs parameter type objects which should use ModelNodeAdapter.  Please try to refactor.  If it becomes too large of an issue, not that important.